### PR TITLE
feat(java-binding): bundle jni library to jar

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

In this PR, we bundle the java binding shared library file in the jar package via the quest db rust plugin. In this way, to run the java binding, we don't have to specially build the rust java binding library, but instead the library can be built during maven build, and the library file can be bundled into the jar of java binding package. The library will be loaded via the quest db jni loader.

With this PR, after the maven project is built, we can handle the stream chunk payload without any extra setting (such as the dynamic library path). Therefore, in this PR, we change to use the stream chunk payload by default in the connector node.

In some environment with no rust build environment being set, we may want to disable building the rust library. We can add a `-Dno-build-rust` option when we run maven command to disable building the rust library.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by ice1000 — COMMENTED** [review]
Wow! QuestDB has a JNI loader!

File: java/java-binding/pom.xml [hunk 1]
```diff
@@ -36,49 +42,54 @@
...
+                <version>${jni.loader.version}</version>
+                <executions>
+                    <execution>
+                        <id>build-jni-rust</id>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                        <configuration>
+                            <path>../../src/java_binding</path>
+                            <release>false</release>
```
Comment (StrikeW) [inline java/java-binding/pom.xml:86]: Is there a way to enable the release build for the java binding crate? (performance concern)

**Review by StrikeW — APPROVED** [review]
Rest lgtm. Thanks for the improvement.

File: java/java-binding/pom.xml (hunk 1 above)
Comment (wenym1) [reply java/java-binding/pom.xml]: Thanks for pointing out! I just added a property `-Djava.binding.release=true` in maven command to enable release build. 

The property has been added to script related to release mode.

**Pull request merged**